File: JavaCoreGoIt/src/module02/HomeWork22By20160822.java
package module02;

public class HomeWork22By20160822 {

    private static double withdrawBalance(double balance, double withdraw, double commission) {
        return balance - withdraw - withdraw * commission / 100;
    }

    private static String withdrawResult(double balance, double withdraw, double commission) {
        double result = withdrawBalance(balance, withdraw, commission);
        if (result >= 0) {
            return "OK " + withdraw * commission / 100 + " " + result;
        } else return "NO";
    }

    public static void main(String[] args) {
        double balance = 250.0;
        double withdraw = 25.0;
        double commission = 5.0;  // in %

        System.out.println(withdrawResult(balance, withdraw, commission));

        balance = 205;
        System.out.println(withdrawResult(balance, withdraw, commission));

        balance = 26;
        System.out.println(withdrawResult(balance, withdraw, commission));

    }

}

File: JavaCoreGoIt/z_kotlin/src/module02/HomeWork22by20160822.kt
package module02

private fun withdrawBalance(balance: Double, withdraw: Double, commission: Double): Double {
    return balance - withdraw - withdraw * commission / 100
}

private fun withdrawResult(balance: Double, withdraw: Double, commission: Double): String {
    val result = withdrawBalance(balance, withdraw, commission)
    if (result >= 0) {
        return "OK " + withdraw * commission / 100 + " " + result
    } else
        return "NO"
}


fun main(args: Array<String>) {
    var balance = 250.0
    val withdraw = 25.0
    val commission = 5.0  // in %

    println(withdrawResult(balance, withdraw, commission))

    balance = 205.0
    println(withdrawResult(balance, withdraw, commission))

    balance = 26.0
    println(withdrawResult(balance, withdraw, commission))

}
